overlord/snapstate/handlers.go

```
@@ -3236,3 +3250,16 @@ func InjectAutoConnect(mainTask *state.Task, snapsup *SnapSetup) {
 	InjectTasks(mainTask, state.NewTaskSet(autoConnect))
 	mainTask.Logf("added auto-connect task")
 }
+
+// GetSnapDirOptions returns *dirs.SnapDirOptions configured according to the
+// enabled experimental features. The state must be locked by the caller.
+func GetSnapDirOptions(state *state.State) *dirs.SnapDirOptions {
+	tr := config.NewTransaction(state)
+
+	hiddenDir, err := features.Flag(tr, features.HiddenSnapFolder)
+	if err != nil {
+		logger.Noticef("cannot read %q conf, will assume it's unset: %s", features.HiddenSnapFolder, err)
```

this should return an error